docker/src/main/java/brooklyn/entity/container/DockerUtils.java

```diff
@@ -224,10 +224,20 @@ public static String imageName(Entity entity, String dockerfile) {
     public static Optional<String> getContainerName(Entity target) {
         return Optional.fromNullable(target.sensors().get(DockerContainer.DOCKER_CONTAINER_NAME))
                 .or(Optional.fromNullable(target.config().get(DockerContainer.DOCKER_CONTAINER_NAME)))
-                .or(Optional.fromNullable(target.config().get(BrooklynCampConstants.PLAN_ID)))
+                .or(Optional.fromNullable(getContainerNameFromPlan(target)))
                 .transform(DockerUtils.ALLOWED);
     }
 
+    private static String getContainerNameFromPlan(Entity target) {
+        String planId = target.config().get(BrooklynCampConstants.PLAN_ID);
+        if (planId != null) {
+            //Plan IDs are not unique even in a single application
+            return planId + "_" + target.getId();
+        } else {
+            return null;
```

Comment:
It would prevent the existing logic from working as expected. If no `PLAN_ID`, then it would use an auto-generated name (see the link in Graeme's post).